### ISSUE TITLE
Purple users conversation & feedback group

### ISSUE THREAD

**Issue by alltheseas** [damus-io/damus#2107]

**Comment by alltheseas** [comment]
cc @jb55

**Comment by alltheseas** [comment]
This would be an useful method for product, design, business team to gather direct feedback from the Damus supporters @danieldaquino 

How might we achieve this?

**Comment by alltheseas** [comment]
Private relay + auth

**Comment by danieldaquino** [comment]
> Private relay + auth

I wonder if there are any out of the box solutions out there for this? I would advise against a custom solution/setup at this point, to avoid spreading ourselves too thin 😅

**Comment by alltheseas** [comment]
> > Private relay + auth
> 
> I wonder if there are any out of the box solutions out there for this? I would advise against a custom solution/setup at this point, to avoid spreading ourselves too thin 😅

> I would advise against a custom solution/setup at this point, to avoid spreading ourselves too thin 😅

:100: 

>  I wonder if there are any out of the box solutions out there for this? I

How about a invite only write permission relay (i.e. like nostr.wine, except gate opens with proof of purple purchase)? Read side can be either public, or private. I like public as it keeps development discussion open, and it helps market purple subscription as a perk in having say where Damus goes next.

**Comment by danieldaquino** [comment]
> How about a invite only write permission relay (i.e. like nostr.wine, except gate opens with proof of purple purchase)? Read side can be either public, or private. I like public as it keeps development discussion open, and it helps market purple subscription as a perk in having say where Damus goes next.

Maybe! I wonder what kind of software/product makes this easy. https://github.com/fiatjaf/khatru comes to mind, but I haven't tried it yet to be sure it fits our needs.

**Comment by alltheseas** [comment]
@MSauce what do yall use for nostr.wine paid relay?

**Comment by jb55** [comment]
from what I understand they have a business-logic proxy in front of a strfry backend

**Comment by jb55** [comment]
to dogfood our own stuff it might be cool to do a rust relay with a nostrdb backend

**Comment by jb55** [comment]
although khatru is definitely an attractive option. from reading the code awhile back it seems to have an LMDB backend as well.

**Comment by alltheseas** [comment]
Which one of these is the least amount of development effort? The least amount of maintenance effort?

**Comment by jb55** [comment]
khatru probably

**Comment by MSauce** [comment]
> from what I understand they have a business-logic proxy in front of a strfry backend

Yes, that's correct though it's primarily for other features like restricting DM metadata, AUTH, NIP-50, etc. 

If all you want to do initially is a pay-to-write relay, you could just use strfry + a simple write plugin (in any language) that checks if the pubkey is a purple subscriber before accepting/rejecting an event. 

You only need something more involved if you want to do read access control/AUTH.

**Comment by alltheseas** [comment]
Goal:

![image](https://github.com/damus-io/damus/assets/64376233/139f098f-2e1f-4541-8f0d-880030f44e0d)

**Comment by alltheseas** [comment]
From mutiny blog: guaranteed support for purple subacribers

> With something like a blinded support system, we could restrict it to paying users only without knowing which user it is and without associating multiple support requests to the same user. All we care about is whether or not the user paid for support tickets at some point and the one-off pubkey they have so we can communicate securely with them. We could allow two monthly support tickets for any subscribed Mutiny+ user, possibly doing one-off paid support tickets for non-subscribers.

https://blog.mutinywallet.com/blinded-authentication/

**Comment by jb55** [comment]
On Thu, May 16, 2024 at 10:23:27AM GMT, alltheseas wrote:
>From mutiny blog: guaranteed support for purple subacribers
>
>> With something like a blinded support system, we could restrict it to paying users only without knowing which user it is and without associating multiple support requests to the same user. All we care about is whether or not the user paid for support tickets at some point and the one-off pubkey they have so we can communicate securely with them. We could allow two monthly support tickets for any subscribed Mutiny+ user, possibly doing one-off paid support tickets for non-subscribers.
>
>https://blog.mutinywallet.com/blinded-authentication/

this is really cool

**Comment by alltheseas** [comment]
Closing as part of a backlog triage — we're narrowing focus to the most actionable issues to reduce noise for contributors.

If this is still relevant, please reopen with updated context or reproduction steps. Thanks for reporting!
